### PR TITLE
Update test_wfm to pass and fix linting issues

### PR DESCRIPTION
**Pull request by nfelt14**
## Proposed changes

This fixes a unit test and fixes a bunch of linting issues in the code.

Addresses #18 

## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] CI/CD update (an update to the CI/CD workflows, scripts, and/or configurations)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have followed the guidelines in the [CONTRIBUTING](https://github.com/tektronix/tm_data_types/blob/main/CONTRIBUTING.md) document
- [x] I have signed the CLA
- [x] I have checked to ensure there aren't other open [Pull Requests](https://github.com/tektronix/tm_data_types/pulls) for the same update/change
- [x] I have created (or updated) an [Issue](https://github.com/tektronix/tm_data_types/issues) to track the status of this update/change and updated the link in this PR description (see above in the **Proposed changes** section) using the wording `Addresses #<issue_number>`
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] Basic linting passes locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added necessary documentation (if appropriate)
- [x] I have updated the Changelog with a brief description of my changes

### CODE REVIEW

File: tests/test_tm_data_types.py [hunk 1]
```diff
@@ -91,50 +83,47 @@ def test_parallel():
...
 
-        values = np.array([10, 11, 12, 32222, 32223, 32224], dtype=np.int16)
+    values = np.array([10, 11, 12, 32222, 32223, 32224], dtype=np.int16)
 
-        waveform.y_axis_values = values
-        waveform.y_axis_spacing = 1 / type_max(np.dtype(np.int16))
-        write_file(waveform_path, waveform)
+    waveform.y_axis_values = values
+    waveform.y_axis_spacing = 1 / type_max(np.dtype(np.int16))
+    waveform.trigger_index = values.size / 2
```
Comment (nfelt14) [inline tests/test_tm_data_types.py:117]: @tekperson, does this seem reasonable to do? To set the trigger point at the halfway mark?

**Review by u625355 — DISMISSED** [review]
Not sure about the trigger index change, except that rest seems fine.
@tekperson , can comment on that.

File: src/tm_data_types/datum/waveforms/digital_waveform.py [hunk 2]
```diff
@@ -57,19 +58,19 @@ def __setattr__(self, key: str, value: Any) -> None:
             key: The attribute name to set as a string.
             value: The value that the attribute is set to.
         """
-        if key in "y_axis_values" "y_axis_spacing" "y_axis_offset":
+        if key in "y_axis_valuesy_axis_spacingy_axis_offset":
```
Comment (qthompso) [inline src/tm_data_types/datum/waveforms/digital_waveform.py:61]: Usually I leave spaces between the searched items just to make it more readable

File: src/tm_data_types/files_and_formats/wfm/data_formats/digital.py [hunk 3]
```diff
@@ -19,7 +19,7 @@ class WaveformFileWFMDigital(WFMFile[DigitalWaveform]):
     ################################################################################################
 
     _META_DATA_LOOKUP = WFMFile.update_bidict(
-        WFMFile._META_DATA_LOOKUP,
+        WFMFile._META_DATA_LOOKUP,  # noqa: SLF001
         {
             "digital_probe_0_state": "d0",
```
Comment (qthompso) [inline src/tm_data_types/files_and_formats/wfm/data_formats/digital.py:24]: Noe: add issue to figure out what d0-d7 even mean. The assumption is they represent whether the digital probe was turned off or on, but I remember that not being the case, so this name may be non representative.

File: src/tm_data_types/files_and_formats/wfm/wfm_format.py [hunk 4]
```diff
@@ -367,16 +385,15 @@ def setup_file_info(
...
         if (
             None not in (self.header, self.pixel_map, self.summary_frame_type)
             or not self.curve_buffer
         ):
             # find the length of the file, along with the curve position
             eof_offset, curve_offset = self._find_offsets()
 
-            # byte offset can be calculated by ignoring the precharge buffer
-            # number of frames can be calculated by checking the length of the curve specs (or update)
+            # byte offset can be calculated by ignoring the pre-charge buffer
```
Comment (qthompso) [inline src/tm_data_types/files_and_formats/wfm/wfm_format.py:395]: In documentation, there is no hyphen for precharge and postcharge

File: tests/test_tm_data_types.py (hunk 1 above)
Comment (qthompso) [reply tests/test_tm_data_types.py]: He mentioned to me that it was what is expected

File: src/tm_data_types/datum/waveforms/digital_waveform.py (hunk 2 above)
Comment (nfelt14) [reply src/tm_data_types/datum/waveforms/digital_waveform.py:61]: Was it supposed to be implicitly concatenated strings or a list?

Comment (qthompso) [reply src/tm_data_types/datum/waveforms/digital_waveform.py:61]: A list

File: src/tm_data_types/files_and_formats/wfm/wfm_format.py [hunk 5]
```diff
@@ -957,19 +986,17 @@ def _find_offsets(self) -> Tuple[int, int]:
                 else:
                     byte_count += len(attribute_value)
             # if attribute is undefined, then we check the length of the type to get a static length
-            elif attribute_name not in "meta_data" "file_checksum":
+            elif attribute_name not in "meta_datafile_checksum":
```
Comment (qthompso) [inline src/tm_data_types/files_and_formats/wfm/wfm_format.py:989]: This should be changed to a list

File: src/tm_data_types/files_and_formats/wfm/wfm_format.py [hunk 6]
```diff
@@ -957,19 +986,17 @@ def _find_offsets(self) -> Tuple[int, int]:
...
-                    try:
+                    with contextlib.suppress(AttributeError):
                         byte_count += get_args(attribute_type)[1].get_cls_length()
-                    except AttributeError:
-                        pass
 
             if (
                 attribute_value is not None
-                and attribute_name not in "file_checksum" "meta_data"
+                and attribute_name not in "file_checksummeta_data"
```
Comment (qthompso) [inline src/tm_data_types/files_and_formats/wfm/wfm_format.py:999]: This should be changed to a list

File: src/tm_data_types/files_and_formats/wfm/data_formats/digital.py (hunk 3 above)
Comment (nfelt14) [reply src/tm_data_types/files_and_formats/wfm/data_formats/digital.py:24]: @qthompso, please go ahead and file an issue

**Review by qthompso — APPROVED** [review]

**Pull request merged**